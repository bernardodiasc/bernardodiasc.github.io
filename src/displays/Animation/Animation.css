.Animation {
  margin-bottom: 20px;
  cursor: pointer;
}

.Animation svg {
  display: block;
}
