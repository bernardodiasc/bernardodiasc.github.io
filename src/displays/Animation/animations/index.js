import * as favouriteAppIcon from './favourite_app_icon.json'
import * as reyUpdated from './rey_updated.json'
import * as legoLoader from './lego_loader.json'

export default {
  favouriteAppIcon,
  reyUpdated,
  legoLoader,
}
