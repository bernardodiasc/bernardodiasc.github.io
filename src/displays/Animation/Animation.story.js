import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import Animation from './Animation'
import AnimationReadme from './README.md'
storiesOf('Displays/Animation', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{AnimationReadme}</Docs>)
  .add('Example', () => <Animation />)
  .add('Minimal render', () => <Animation />)
