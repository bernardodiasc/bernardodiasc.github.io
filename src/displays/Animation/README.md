# Animation

## How to use

```
import Animation from 'displays/Animation'
```

```
<Animation />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
