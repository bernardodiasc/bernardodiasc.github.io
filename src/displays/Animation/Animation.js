import React, { PureComponent } from 'react'
import Lottie from 'react-lottie'
import animations from './animations'
import './Animation.css'

class Animation extends PureComponent {
  static defaultProps = {
    animation: '',
    width: '100%',
    height: '100%',
    loop: true,
    autoPlay: true,
  }

  constructor(props) {
    super(props)
    this.state = {
      isStopped: !this.props.autoPlay,
      isPaused: !this.props.autoPlay,
      isComplete: false,
    }
  }

  handleEvent = (obj) => {
    if (!this.props.loop) {
      if (obj.currentTime === (obj.totalTime - 1)) {
        if (this.state.isComplete) {
          this.setState({ isStopped: true, isComplete: false })
        } else {
          this.setState({ isStopped: false, isComplete: true })
        }
      }
    }
  }

  render() {
    const animation = animations[this.props.animation]
    const defaultOptions = {
      loop: this.props.loop,
      autoplay: this.props.autoPlay,
      animationData: animation,
      rendererSettings: {
        preserveAspectRatio: 'xMidYMid slice',
      }
    }
    const makeValidNumber = (value) =>
      value.substr(value.length - 1) === '%' ? value : Number(value)

    return (
      <div className="Animation">
        <Lottie
          onClick={() => {}}
          options={defaultOptions}
          width={makeValidNumber(this.props.width)}
          height={makeValidNumber(this.props.height)}
          isStopped={this.state.isStopped}
          isPaused={this.state.isPaused}
          eventListeners={
            [
              {
                eventName: 'enterFrame',
                callback: obj => this.handleEvent(obj),
              },
            ]
          }
        />
      </div>
    )
  }
}

export default Animation
