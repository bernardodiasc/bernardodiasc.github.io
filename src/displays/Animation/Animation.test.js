import React from 'react'
import { shallow } from 'enzyme'

import Animation from './Animation.js'

it('renders without crashing', () => {
  const component = shallow(<Animation />)
  expect(component).toMatchSnapshot()
})
