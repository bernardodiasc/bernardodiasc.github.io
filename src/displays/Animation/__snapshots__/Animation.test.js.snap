// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Animation"
>
  <Lottie
    ariaLabel="animation"
    ariaRole="button"
    eventListeners={
      Array [
        Object {
          "callback": [Function],
          "eventName": "enterFrame",
        },
      ]
    }
    height="100%"
    isClickToPauseDisabled={false}
    isPaused={false}
    isStopped={false}
    onClick={[Function]}
    options={
      Object {
        "animationData": undefined,
        "autoplay": true,
        "loop": true,
        "rendererSettings": Object {
          "preserveAspectRatio": "xMidYMid slice",
        },
      }
    }
    speed={1}
    title=""
    width="100%"
  />
</div>
`;
